pkg/azurefile/controllerserver.go

```diff
@@ -212,7 +215,51 @@ func (d *Driver) ControllerUnpublishVolume(ctx context.Context, req *csi.Control
 
 // CreateSnapshot create a snapshot (todo)
 func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	klog.V(2).Infof("CreateSnapshot called with request %v", *req)
+
+	sourceVolumeID := req.GetSourceVolumeId()
+	snapshotName := req.Name
+	if len(snapshotName) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Snapshot name must be provided")
+	}
+	if len(sourceVolumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "CreateSnapshot Source Volume ID must be provided")
```

Comment:
source volume id is empty in CreateSnapshot

Reply:
Source volume id is necessary. Otherwise, we can not know which FileShare the snapshot is created from.